libfwupdplugin/fu-efivar-freebsd.c

```diff
@@ -0,0 +1,396 @@
+ *
+ * Since: 1.6.1
+ **/
+gboolean
+fu_efivar_set_data (const gchar *guids, const gchar *name, const guint8 *data,
+		     gsize sz, guint32 attr, GError **error)
+{
+	efi_guid_t guid;
+	efi_str_to_guid (guids, &guid);
+
+	if (efi_set_variable (guid, name, data, sz, attr) != 0) {
+		g_prefix_error (error, "failed to write data to efivar: ");
+		return FALSE;
+	}
+
+	/* success */
+	return TRUE;
+}
+
+/**
+ * fu_efivar_delete:
+ * @guid: Globally unique identifier
+ * @name: Variable name
+ * @error: #GError
+ *
+ * Removes a variable from NVRAM
+ *
+ * Returns: %TRUE on success
+ *
+ * Since: 1.6.1
+ **/
+gboolean
+fu_efivar_delete (const gchar *guids, const gchar *name, GError **error)
+{
+	efi_guid_t guid;
+	efi_str_to_guid (guids, &guid);
+
+	if (efi_del_variable (guid, name) == 0)
+		return TRUE;
+	return FALSE;
+}
+
+/**
+ * fu_efivar_delete_with_glob:
+ * @guid: Globally unique identifier
+ * @name_glob: Variable name
+ * @error: #GError
+ *
+ * Removes a group of variables from NVRAM
+ *
+ * Returns: %TRUE on success
+ *
+ * Since: 1.6.1
+ **/
+gboolean
+fu_efivar_delete_with_glob (const gchar *guids, const gchar *name_glob, GError **error)
+{
+	efi_guid_t guid_to_delete;
+	efi_str_to_guid (guids, &guid_to_delete);
+
+	efi_guid_t *guid = NULL;
+	gchar *name = NULL;
+
+	gboolean rv = FALSE;
+
+	while (efi_get_next_variable_name (&guid, &name)) {
+		if (memcmp (&guid_to_delete, guid, sizeof (guid)) != 0)
+			continue;
+		if (!fu_common_fnmatch (name, name_glob))
+			continue;
+		rv = fu_efivar_delete (guids, name, error);
+		if (!rv)
+			break;
+	}
+	return rv;
+}
+
+static gboolean
+fu_efivar_exists_guid (const gchar *guids)
+{
+	efi_guid_t *guid = NULL;
+	gchar *name = NULL;
+	efi_guid_t test;
+	efi_str_to_guid (guids, &test);
+
+	while (efi_get_next_variable_name (&guid, &name)) {
+		if (memcmp (&test, guid, sizeof (test)) == 0) {
+			return TRUE;
+		}
+	}
+	return FALSE;
+}
+
+/**
+ * fu_efivar_exists:
+ * @guid: Globally unique identifier
+ * @name: (nullable): Variable name
+ *
+ * Test if a variable exists
+ *
+ * Returns: %TRUE on success
+ *
+ * Since: 1.6.1
+ **/
+gboolean
+fu_efivar_exists (const gchar *guids, const gchar *name)
+{
+	g_return_val_if_fail (guids != NULL, FALSE);
+
+	/* any name */
+	if (name == NULL)
+		return fu_efivar_exists_guid (guids);
+
+	return fu_efivar_get_data (guids, name, NULL, NULL, NULL, NULL);
+}
+
+/**
+ * fu_efivar_get_data:
+ * @guid: Globally unique identifier
+ * @name: Variable name
+ * @data: Data to set
+ * @data_sz: size of data
+ * @attr: Attributes
+ * @error: (nullable): optional return location for an error
+ *
+ * Gets the data from a UEFI variable in NVRAM
+ *
+ * Returns: %TRUE on success
+ *
+ * Since: 1.6.1
+ **/
+gboolean
+fu_efivar_get_data (const gchar *guids, const gchar *name, guint8 **data,
+		       gsize *data_sz, guint32 *attr, GError **error)
+{
+	efi_guid_t guid;
+	efi_str_to_guid (guids, &guid);
+
+	return !(efi_get_variable (guid, name, data, data_sz, attr))
```

Comment:
remove outer brackets perhaps? or check for != 0 perhaps?

Reply:
Changed to a `!= 0` check.